### ISSUE TITLE
v1 version of the snowflake api's being deprecated.

### ISSUE THREAD

**Issue by eegasai** [fixer-m/snowflake-db-net-client#43]
Hi,

I am using this package to query snowflake data. Recently I have come across one article where it is mentioned that the api's used as part of this package like

/ session/v1/login-request
/queries/v1/query-request

will be turned off.

Snowflake team plans to move to the V2+ version of the API's (i think may be to SQL REST API).

So just wanted to check that, you are aware of any such changes or is there anything in this package roadmap to change the API's?

Reference: [Article](https://community.snowflake.com/s/question/0D5Do00000O3Nh1KAF/i-would-like-help-understanding-snowflakes-rest-api-version-progression)

**Comment by julealgon** [comment]
This post just made me reconsider using this library at all. Rolling the code back as I type this. I might end up rolling my own client to consume the v2 REST API.

**Comment by georgebarbu** [comment]
Any news on this? Is a migration to Snowflake official V2 API even possible?

**Comment by tom-j-irvine** [comment]
I needed to implement my own simple client using the v2 apis - the primary thing that will be different is that v2 only supports OAuth or Key Pair authentication (no basic user/pass which is being used here): https://docs.snowflake.com/en/developer-guide/sql-api/authenticating

I contributed a bit of code to an issue about supporting key/pair auth that is similar to what I'm using for my client here:  https://github.com/fixer-m/snowflake-db-net-client/issues/16

Other than that, the v2 api is pretty straight-forward and seems to work in a similar manner.  My requirements were quite limited, so what I have probably won't work for most people, but I'm happy to share what I have if it helps.

**Comment by fixer-m** [comment]
@eegasai Yes, I heard about API v2 a long time ago. Now I see it has pretty good documentation and I guess it has most of features from v1. So this client will also move to this API, I'll try to do this without breaking changes. Thanks!

**Comment by prluk** [comment]
@fixer-m do you an ETA for the API v2 transition?
